### PR TITLE
Refactor/todo page

### PR DESCRIPTION
**Pull request by nhattlm3173**
## ✅ Testing Checklist (MUST be completed before merge)

- [ ] Unit tests added or updated
- [ ] Integration/API tests covered
- [ ] Manual test completed
- [ ] All edge cases and failure paths tested
- [ ] No regressions or breaking changes
- [ ] Test coverage acceptable

## 📷 Screenshots / Recordings (if UI-related)

---

> ⚠️ **Please do not request a review or merge until all relevant boxes are checked.**

### CODE REVIEW

File: src/component/Form/index.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ function Form({
   validationSchema,
   submitMode = SubmitMode.onSubmitButton,
   defaultValues = {},
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (minhkhanb) [inline src/component/Form/index.tsx:15]: Remove rule

File: src/component/Form/index.tsx [hunk 2]
```diff
@@ -22,10 +23,10 @@ function Form({
     defaultValues,
   });
 
-  console.log(_ ? 'true' : 'false');
-
   const formRef = useRef<HTMLFormElement>(null);
-  const [formId] = useId();
+  const formId = useId();
+
+  // console.log(formId);
```
Comment (minhkhanb) [inline src/component/Form/index.tsx:29]: Remove log

File: src/component/Form/types/IForm.ts [hunk 3]
```diff
@@ -14,7 +14,13 @@ export interface FormProps {
...
     defaultValues?: Record<string, unknown>,
     formState?: FormState<Record<string, unknown>>,
-    formHandlers?: unknown
+    formHandlers?: UseFormReturn<
+      Record<string, unknown>,
+      unknown,
+      {
+        [x: string]: unknown;
+      }
+    >
```
Comment (minhkhanb) [inline src/component/Form/types/IForm.ts:23]: Avoid unknow type when use generic type

File: src/component/Form/types/IForm.ts [hunk 4]
```diff
@@ -28,7 +34,13 @@ export interface FormOptions {
...
```
Comment (minhkhanb) [inline src/component/Form/types/IForm.ts:43]: The same above

File: src/section/Test/components/TestForm/index.tsx [hunk 5]
```diff
@@ -19,7 +19,13 @@ export default function TestForm() {
...
         defaultValues?: Record<string, unknown>,
         formState?: FormState<Record<string, unknown>>,
-        formHandlers?: unknown
+        formHandlers?: UseFormReturn<
+          Record<string, unknown>,
+          unknown,
+          {
+            [x: string]: unknown;
+          }
+        >
```
Comment (minhkhanb) [inline src/section/Test/components/TestForm/index.tsx:28]: The same above

File: src/section/Todo/components/AddTodoFormFields/index.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  todoSelectedValue: string;
+  todoToUpdate: TodoValue | null;
+  setTodoToUpdateAction: Dispatch<SetStateAction<TodoValue | null>>;
+  setTodoSelectedValue: React.Dispatch<React.SetStateAction<string>>;
+}) {
+  const form = useFormContext();
+
+  useEffect(() => {
+    form.setValue('message', todoSelectedValue);
+  }, [todoSelectedValue, form]);
```
Comment (minhkhanb) [inline src/section/Todo/components/AddTodoFormFields/index.tsx:23]: Consider binding message value through on Form component
